apps/examples/camera/camera_main.c

```diff
@@ -227,6 +302,153 @@ static void free_buffer(struct v_buffer *buffers, uint8_t bufnum)
 	}
 }
 
+void clear_camera_options()
+{
+	int i = 0;
+	int size = sizeof(ctrls) / sizeof(struct camera_ctrls) - 1;
+	do {
+		ctrls[i].active = 0;
+		ctrls[i].neg = 0;
+		ctrls[i].req = 0;
+		ctrls[i].value = 0;
+		ctrls[i].min = 0;
+	} while (i++ < size);
+}
+
+void get_camera_options(int fd, int flag)
+{
+	int i = 0;
+	int j = 0;
+	int ret = 0;
+	int size = sizeof(ctrls) / sizeof(struct camera_ctrls) - 1;
+	struct v4l2_queryctrl query;
+	struct v4l2_fmtdesc fmtdesc;
+	struct v4l2_frmsizeenum fsize;
+	char buff[128];
+
+	if (flag) {
+		fprintf(stdout, "Supported Camera Options :\n");
+	}
+	/* Control Options */
+	do {
+		query.id = ctrls[i].v4l2_id;
+		query.ctrl_class = ctrls[i].v4l2_class;
+		ret = ioctl(fd, VIDIOC_QUERYCTRL, &query);
+		ctrls[i].active = 0;
+		if (ret >= 0) {
+			ctrls[i].active = 1;
+			ctrls[i].neg = 0;
+			if (query.minimum < 0) {
+				ctrls[i].neg = 1;
+				ctrls[i].min = query.minimum * -1;
+				query.maximum += query.minimum * -1;
+				query.minimum = 0;
+				if (query.default_value < 0) {
+					query.default_value += (query.default_value * -1);
+				}
+			}
+			if (flag) {
+				sprintf(buff, ctrls[i].sbuff, query.minimum, query.maximum, query.step, query.default_value);
+				fprintf(stdout, buff);
+			}
+		}
+	} while (i++ < size);
+	/* Format options */
+	i = 0;
+	if (flag) {
+		fprintf(stdout, SUPPORTED_FORMAT);
+	}
+	do {
+		fmtdesc.index = i;
+		fmtdesc.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+		ret = ioctl(fd, VIDIOC_ENUM_FMT, &fmtdesc);
+		if (ret < 0) {
+			continue;
```

Comment:
normally, if ioctl returns negative value, it means error. so break & return error is correct, IMO.

Reply:
Only for ENUM_FMT we are not returning the control. as the indexes are not sequential.